### DISCUSSION TITLE
Data connector for RAG

### DISCUSSION THREAD

**Discussion by flefevre** [open-webui/open-webui#3226] [Ideas]
Dear all,

would it be possible to add specific data connector for RAG?
I am looking to add document from :

- youtube
- https://github.com/
- https://docs.tuleap.org/user-guide/code-versioning/git.html

The two last are really important since they enable to connect to git repositories

Thanks for your idea

**Comment by flefevre** [comment]
It will be also pertinent to add connector for research scientist 

Typically to scientific paper databases such as

-pubmed https://pubmed.ncbi.nlm.nih.gov/
-hal https://hal.science/
-zotero https://www.zotero.org/ used by researcher to save their bibliography 

I am convinced that success of open WebUI will come by offering llm access but also public data access.

Is there any comment ?

**Comment by joelmnz** [comment]
I know that [Microsoft Azure AI Search](https://azure.microsoft.com/en-us/products/ai-services/ai-search/) is used in the corporate area, if you could plug something like that in it would open up a world of possibility for businesses wanting to use Open WebUI

Also something like Notion which as API access as this could have a large personal user knowledge base to pull from.

**Comment by flefevre** [comment]
A connector to 'sharepoint' and Google drive system could be also of great interest.
I do not know if it should be the work of 'open webui team' or the job of the embedded database chroma team?

**Comment by jfbloom22** [comment]
I would love to support my coworkers with the ability to put documents into a SharePoint folder, select that folder in WebUI as the knowledge base.  Each time a document is changed, or added, it is re-embedded.  
I am open to working on this, but I need some guidance on where something like this should be added.  Should I try to extend WebUI?  Should I try to get this working in the form of a function or pipe?

**Reply by flefevre** [reply]
And if the SharePoint folder is vectorized then the vector tenant should have the same right on it .

**Reply by rgaricano** [reply]
Maybe this help u: https://github.com/open-webui/cookbook/blob/main/knowledge/add-to-knowledge.ipynb

**Reply by jfbloom22** [reply]
@rgaricano this is a great start!   Thanks for pointing this out.  Great demonstration of the WebUI API endpoints for managing the knowledge base. 

Looks like I can have Sharepoint call these API endpoints any time a file changes:  https://learn.microsoft.com/en-us/archive/technet-wiki/38062.sharepoint-online-simulate-sharepoint-timer-jobs-and-event-receivers-using-azure-functions-and-sharepoint-webhooks#SharePoint_Webhooks_as_Event_Receivers

**Comment by martikevin** [comment]
A lot of documentation in large enterprises is on Sharepoint and Confluence. Would be nice if one could access this data as an external RAG.
